QT 6.9.0 Unable To Build

# System Information
* Arch Linux
* GCC 14.2.1+r753+g1cd744a6828f
* glibc 2.41+r9+ga900dbaf70f0-1
* qt6 6.9.0

# Error
```
[127/135] Building CXX object mcpelauncher-ui-qt/CMakeFiles/mcpelauncher-ui-qt.dir/launcherapp.cpp.o
FAILED: mcpelauncher-ui-qt/CMakeFiles/mcpelauncher-ui-qt.dir/launcherapp.cpp.o 
/usr/bin/c++ -DLAUNCHER_DISABLE_DEV_MODE -DLAUNCHER_ENABLE_GLFW -DLAUNCHER_ENABLE_GOOGLE_PLAY_LICENCE_CHECK -DLAUNCHER_MACOS_HAVE_ARMLAUNCHER -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_POSITIONING_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMETA_LIB -DQT_QMLMODELS_LIB -DQT_QMLWORKERSCRIPT_LIB -DQT_QML_LIB -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WEBCHANNEL_LIB -DQT_WEBENGINECORE_LIB -DQT_WEBENGINEWIDGETS_LIB -DQT_WIDGETS_LIB -I/home/hurricane/aur/mcpelauncher-ui/src/build/mcpelauncher-ui-qt -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-ui-qt -I/home/hurricane/aur/mcpelauncher-ui/src/build/mcpelauncher-ui-qt/mcpelauncher-ui-qt_autogen/include -I/home/hurricane/aur/mcpelauncher-ui/src/build/_deps/glfw3_ext-src/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/playdl-signin-ui-qt/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-extract/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-common/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-apkinfo/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/axml-parser/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/file-util/include -I/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/google-play-api/include -I/home/hurricane/aur/mcpelauncher-ui/src/build/google-play-api -isystem /usr/include/qt6/QtCore -isystem /usr/include/qt6 -isystem /usr/lib/qt6/mkspecs/linux-g++ -isystem /usr/include/qt6/QtQuick -isystem /usr/include/qt6/QtGui -isystem /usr/include/qt6/QtQml -isystem /usr/include/qt6/QtQmlIntegration -isystem /usr/include/qt6/QtNetwork -isystem /usr/include/qt6/QtQmlMeta -isystem /usr/include/qt6/QtQmlModels -isystem /usr/include/qt6/QtQmlWorkerScript -isystem /usr/include/qt6/QtOpenGL -isystem /usr/include/qt6/QtSvg -isystem /usr/include/qt6/QtConcurrent -isystem /usr/include/qt6/QtWidgets -isystem /usr/include/qt6/QtWebEngineWidgets -isystem /usr/include/qt6/QtWebEngineCore -isystem /usr/include/qt6/QtWebChannel -isystem /usr/include/qt6/QtPositioning -isystem /usr/include/qt6/QtPrintSupport -march=znver3 -O3 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection         -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wp,-D_GLIBCXX_ASSERTIONS -flto=auto -DNDEBUG -std=gnu++17 -mno-direct-extern-access -MD -MT mcpelauncher-ui-qt/CMakeFiles/mcpelauncher-ui-qt.dir/launcherapp.cpp.o -MF mcpelauncher-ui-qt/CMakeFiles/mcpelauncher-ui-qt.dir/launcherapp.cpp.o.d -o mcpelauncher-ui-qt/CMakeFiles/mcpelauncher-ui-qt.dir/launcherapp.cpp.o -c /home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-ui-qt/launcherapp.cpp
/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-ui-qt/launcherapp.cpp: In member function ‘int LauncherApp::launchProfileFile(QString, QString, bool)’:
/home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-ui-qt/launcherapp.cpp:85:155: error: no matching function for call to ‘QChar::QChar(bool&)’
   85 |     if(!trialMode && m_settings.value("latest_version_id").toString() != (m_latestVersion + QChar((char)m_latestVersionCode) + QChar(m_latestVersionIsBeta))) {
      |                                                                                                                                                           ^
In file included from /usr/include/qt6/QtCore/qstring.h:14,
                 from /usr/include/qt6/QtCore/qcoreapplication.h:8,
                 from /usr/include/qt6/QtWidgets/qapplication.h:8,
                 from /usr/include/qt6/QtWidgets/QApplication:1,
                 from /home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-ui-qt/launcherapp.h:4,
                 from /home/hurricane/aur/mcpelauncher-ui/src/mcpelauncher-ui-manifest/mcpelauncher-ui-qt/launcherapp.cpp:1:
/usr/include/qt6/QtCore/qchar.h:149:43: note: candidate: ‘template<class Char, typename std::enable_if<std::is_same<Char, unsigned char>::value, bool>::type <anonymous> > constexpr QChar::QChar(Char)’
  149 |     QT_ASCII_CAST_WARN constexpr explicit QChar(const Char c) noexcept : ucs(c) { }
      |                                           ^~~~~
/usr/include/qt6/QtCore/qchar.h:149:43: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qchar.h:148:66: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  148 |     template <typename Char, if_explicit_ascii_warn_char<Char> = true>
      |                                                                  ^~~~
/usr/include/qt6/QtCore/qchar.h:143:45: note: candidate: ‘template<class Char, typename std::enable_if<std::is_same<Char, char>::value, bool>::type <anonymous> > constexpr QChar::QChar(Char)’
  143 |     QT_ASCII_CAST_WARN constexpr Q_IMPLICIT QChar(const Char ch) noexcept : ucs(uchar(ch)) {}
      |                                             ^~~~~
/usr/include/qt6/QtCore/qchar.h:143:45: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qchar.h:142:66: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  142 |     template <typename Char, if_implicit_ascii_warn_char<Char> = true>
      |                                                                  ^~~~
/usr/include/qt6/QtCore/qchar.h:134:24: note: candidate: ‘template<class Char, typename std::enable_if<std::disjunction<std::is_same<Char, char32_t>, std::conjunction<std::is_same<Char, wchar_t>, std::integral_constant<bool, true> >, std::is_same<Char, int>, std::is_same<Char, unsigned int> >::value, bool>::type <anonymous> > constexpr QChar::QChar(Char)’
  134 |     constexpr explicit QChar(const Char ch) noexcept
      |                        ^~~~~
/usr/include/qt6/QtCore/qchar.h:134:24: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qchar.h:133:66: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  133 |     template <typename Char, if_explicit_conversion_char<Char> = true>
      |                                                                  ^~~~
/usr/include/qt6/QtCore/qchar.h:132:26: note: candidate: ‘template<class Char, typename std::enable_if<std::disjunction<std::is_same<Char, short unsigned int>, std::is_same<Char, short int>, std::is_same<Char, QChar::SpecialCharacter>, std::is_same<Char, QLatin1Char>, std::conjunction<std::is_same<Char, wchar_t>, std::negation<std::integral_constant<bool, true> > >, std::is_same<Char, char16_t> >::value, bool>::type <anonymous> > constexpr QChar::QChar(Char)’
  132 |     constexpr Q_IMPLICIT QChar(const Char ch) noexcept : ucs(char16_t(ch)) {}
      |                          ^~~~~
/usr/include/qt6/QtCore/qchar.h:132:26: note:   template argument deduction/substitution failed:
/usr/include/qt6/QtCore/qchar.h:131:66: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  131 |     template <typename Char, if_implicit_conversion_char<Char> = true>
      |                                                                  ^~~~
/usr/include/qt6/QtCore/qchar.h:119:24: note: candidate: ‘constexpr QChar::QChar(uchar, uchar)’
  119 |     constexpr explicit QChar(uchar c, uchar r) noexcept : ucs(char16_t((r << 8) | c)) {}
      |                        ^~~~~
/usr/include/qt6/QtCore/qchar.h:119:24: note:   candidate expects 2 arguments, 1 provided
/usr/include/qt6/QtCore/qchar.h:115:26: note: candidate: ‘constexpr QChar::QChar()’
  115 |     constexpr Q_IMPLICIT QChar() noexcept : ucs(0) {}
      |                          ^~~~~
/usr/include/qt6/QtCore/qchar.h:115:26: note:   candidate expects 0 arguments, 1 provided
/usr/include/qt6/QtCore/qchar.h:48:31: note: candidate: ‘constexpr QChar::QChar(const QChar&)’
   48 | class QT6_ONLY(Q_CORE_EXPORT) QChar {
      |                               ^~~~~
/usr/include/qt6/QtCore/qchar.h:48:31: note:   no known conversion for argument 1 from ‘bool’ to ‘const QChar&’
/usr/include/qt6/QtCore/qchar.h:48:31: note: candidate: ‘constexpr QChar::QChar(QChar&&)’
/usr/include/qt6/QtCore/qchar.h:48:31: note:   no known conversion for argument 1 from ‘bool’ to ‘QChar&&’
[133/135] Building CXX object mcpelauncher-ui-qt/CMakeFiles/mcpelauncher-ui-qt.dir/mcpelauncher-ui-qt_autogen/mocs_compilation.cpp.o
```

--- a/googleversionchannel.cpp
+++ b/googleversionchannel.cpp
@@ -72,7 +72,7 @@ void GoogleVersionChannel::onAppInfoReceived(const QString &packageName, const Q
             } else {
                 this->m_hasVerifiedLicense |= hasVerifiedLicense;
                 licenseStatus = hasVerifiedLicense ? GoogleVersionChannelLicenceStatus::SUCCEDED : GoogleVersionChannelLicenceStatus::FAILED;
-                m_settings.setValue("latest_version_id", hasVerifiedLicense ? (m_latestVersion + QChar((char)m_latestVersionCode) + QChar(m_latestVersionIsBeta)) : "");
+                m_settings.setValue("latest_version_id", hasVerifiedLicense ? (m_latestVersion + QChar((char)m_latestVersionCode) + QChar((char)m_latestVersionIsBeta)) : "");
             }
             statusChanged();
         });
@@ -89,7 +89,7 @@ void GoogleVersionChannel::onAppInfoFailed(QString const& packageName, const QSt
             if(!trialMode) {
                 m_settings.setValue("latest_version_id", "");
             }
-        } else if(trialMode || m_settings.value("latest_version_id").toString() == (m_latestVersion + QChar((char)m_latestVersionCode) + QChar(m_latestVersionIsBeta))) {
+        } else if(trialMode || m_settings.value("latest_version_id").toString() == (m_latestVersion + QChar((char)m_latestVersionCode) + QChar((char)m_latestVersionIsBeta))) {
             m_hasVerifiedLicense = true;
             licenseStatus = GoogleVersionChannelLicenceStatus::OFFLINE;
         }

--- a/launcherapp.cpp
+++ b/launcherapp.cpp
@@ -82,7 +82,7 @@ int LauncherApp::launchProfileFile(QString profileName, QString filePath, bool s
     auto m_latestVersion = m_settings.value("latest_version").toString();
     auto m_latestVersionCode = m_settings.value("latest_version_code").toInt();
     auto m_latestVersionIsBeta = m_settings.value("latest_version_isbeta").toBool();
-    if(!trialMode && m_settings.value("latest_version_id").toString() != (m_latestVersion + QChar((char)m_latestVersionCode) + QChar(m_latestVersionIsBeta))) {
+    if(!trialMode && m_settings.value("latest_version_id").toString() != (m_latestVersion + QChar((char)m_latestVersionCode) + QChar((char)m_latestVersionIsBeta))) {
         printf("Something went wrong\n");
         shouldExit(1);
     }

